File: reference-code/src/main/java/createcommands/arguments/types/misc/EnchantmentArguments.java
package createcommands.arguments.types.misc;

import dev.jorel.commandapi.CommandAPICommand;
import dev.jorel.commandapi.arguments.EnchantmentArgument;
import dev.jorel.commandapi.arguments.IntegerArgument;
import org.bukkit.enchantments.Enchantment;

class EnchantmentArguments {
    static {
        // #region enchantmentArgumentsExample
        new CommandAPICommand("enchantitem")
            .withArguments(new EnchantmentArgument("enchantment"))
            .withArguments(new IntegerArgument("level", 1, 5))
            .executesPlayer((player, args) -> {
                Enchantment enchantment = (Enchantment) args.get("enchantment");
                int level = (int) args.get("level");

                // Add the enchantment
                player.getInventory().getItemInMainHand().addEnchantment(enchantment, level);
            })
            .register();
        // #endregion enchantmentArgumentsExample
    }
}

File: reference-code/src/main/kotlin/createcommands/arguments/types/misc/EnchantmentArguments.kt
package createcommands.arguments.types.misc

import dev.jorel.commandapi.CommandAPICommand
import dev.jorel.commandapi.arguments.EnchantmentArgument
import dev.jorel.commandapi.arguments.IntegerArgument
import dev.jorel.commandapi.executors.PlayerCommandExecutor
import dev.jorel.commandapi.kotlindsl.commandAPICommand
import dev.jorel.commandapi.kotlindsl.enchantmentArgument
import dev.jorel.commandapi.kotlindsl.integerArgument
import dev.jorel.commandapi.kotlindsl.playerExecutor
import org.bukkit.enchantments.Enchantment

fun enchantmentArguments() {
    // #region enchantmentArgumentsExample
    CommandAPICommand("enchantitem")
        .withArguments(EnchantmentArgument("enchantment"))
        .withArguments(IntegerArgument("level", 1, 5))
        .executesPlayer(PlayerCommandExecutor { player, args ->
            val enchantment = args["enchantment"] as Enchantment
            val level = args["level"] as Int

            // Add the enchantment
            player.inventory.itemInMainHand.addEnchantment(enchantment, level)
        })
        .register()
    // #endregion enchantmentArgumentsExample
}

fun enchantmentArgumentsDSL() {
    // #region enchantmentArgumentsExampleDSL
    commandAPICommand("enchantitem") {
        enchantmentArgument("enchantment")
        integerArgument("level", 1, 5)
        playerExecutor { player, args ->
            val enchantment = args["enchantment"] as Enchantment
            val level = args["level"] as Int

            // Add the enchantment
            player.inventory.itemInMainHand.addEnchantment(enchantment, level)
        }
    }
    // #endregion enchantmentArgumentsExampleDSL
}